src/index.ts

```diff
@@ -69,17 +71,24 @@ async function retryFetch(
   jsonFetchOptions: JsonFetchOptions,
 ): Promise<Response> {
   const shouldRetry = jsonFetchOptions.shouldRetry ?? DEFAULT_SHOULD_RETRY;
+  const onRequestStart =
+    typeof jsonFetchOptions.onRequestStart === 'function' ? jsonFetchOptions.onRequestStart : null;
+  const onRequestEnd =
+    typeof jsonFetchOptions.onRequestEnd === 'function' ? jsonFetchOptions.onRequestEnd : null;
   const retryOptions = {...DEFAULT_RETRY_OPTIONS, ...jsonFetchOptions.retry};
   const requestOptions = getRequestOptions(jsonFetchOptions);
 
   try {
     const response = await promiseRetry(async (throwRetryError, retryCount) => {
+      if (onRequestStart !== null) onRequestStart(requestUrl, retryCount);
```

Comment:
Potentially we can do:
```suggestion
      jsonFetchOptions.onRequestStart?.(requestUrl, retryCount);
```
This will only run the function if its not undefined.